data/modules/BreakdownServicing/BreakdownServicing.lua

```diff
@@ -244,32 +244,41 @@ end
 
 local onEnterSystem = function (ship)
 	if ship:IsPlayer() then
-		print(('DEBUG: Jumps since warranty: %d, chance of failure (if > 0): 1/%d\nWarranty expires: %s'):format(service_history.jumpcount,max_jumps_unserviced-service_history.jumpcount,Format.Date(service_history.lastdate + service_history.service_period)))
+		print(('DEBUG: Jumps since warranty: %d\nWarranty expires: %s'):format(service_history.jumpcount,Format.Date(service_history.lastdate + service_history.service_period)))
 	else
 		return -- Don't care about NPC ships
 	end
-	local saved_by_this_guy = savedByCrew(ship)
-	if (service_history.lastdate + service_history.service_period < Game.time) and not saved_by_this_guy then
+
+	-- Jump drive is getting worn and is running down
+	if (service_history.lastdate + service_history.service_period < Game.time) then
 		service_history.jumpcount = service_history.jumpcount + 1
-		if (service_history.jumpcount > max_jumps_unserviced) or (Engine.rand:Integer(max_jumps_unserviced - service_history.jumpcount) == 0) then
+	end
+
+	-- Test if the engine will actually break
+	if Engine.rand:Number() < service_history.jumpcount / max_jumps_unserviced then
+		print("Player hyperdrive is breaking down!")
+
+		-- The more engineers the more likely to be saved (rule doesn't apply to cooking):
+		local is_saved_by_crew = savedByCrew(ship)
+
+		if is_saved_by_crew then
+			-- Brag to the player
+			if is_saved_by_crew.player then
```

Comment:
If it is a bool, then why does it have an attribute? ;-)

Reply:
Because, "oops"? I'm half right though, because it's `false` if no helping crew is found.

Now called `crew_to_the_rescue`. Hope that's better. Updated.